source/s3_meta_request.c

```diff
@@ -1355,7 +1357,7 @@ static int s_s3_meta_request_incoming_body(
     }
 
     if (request->send_data.response_body.capacity == 0) {
-        if (request->has_part_size_response_body) {
+        if (request->has_part_size_response_body && successful_response) {
```

Comment:
Ignore the buffer ticket, as the error response might not fit into the originally reserved buffer.